### PR TITLE
fix: reanchor pi status after session replacement

### PR DESCRIPTION
**Pull request by dmmulroy**
## Summary

- forward Pi session lifecycle reasons with session reports
- reanchor full-lifecycle authority for Pi startup, new, resume, and fork transitions
- allow an explicitly resumed Pi session to leave the stale-session set while keeping the abandoned session stale
- bump the Pi integration to v5 and verify bundled asset versions match their runtime constants

Without this, Herdr can acknowledge Pi reports after a session replacement while retaining the previous JSONL path, leaving a visibly working Pi pane reported as idle or done.

## Tests

- `just ci`
- focused session replacement, stale resume, negative lifecycle, and v4 upgrade tests
- adversarial pre-fix reproduction and post-fix validation

refs #943

### CODE REVIEW

**Comment by dmmulroy** [comment]
i will conquer the pi integration and all of the edge cases one of these times

File: src/integration/assets/pi/herdr-agent-state.ts [hunk 1]
```diff
@@ -336,13 +337,13 @@ export default function (pi) {
...
 
-  pi.on("session_start", (_event, ctx) => {
+  pi.on("session_start", (event, ctx) => {
     if (ctx?.hasUI !== true) {
       return;
     }
     rootSession = true;
     updateSessionRef(ctx);
-    void reportSession();
+    void reportSession(event?.reason);
```
Comment (ogulcancelik) [inline src/integration/assets/pi/herdr-agent-state.ts]: hmm could this race with below `publishState(true);`? since we don't wait it, state update could arrive first then herdr would drop reporting the new session again. maybe we should wait this?

**Comment by ogulcancelik** [comment]
> i will conquer the pi integration and all of the edge cases one of these times

pi went from one of the easiest agents to build an integration on top of to one of the most edge case heavy pretty fast :/ but I appreciate the help!

File: src/integration/assets/pi/herdr-agent-state.ts (hunk 1 above)
Comment (dmmulroy) [reply src/integration/assets/pi/herdr-agent-state.ts]: fixed 🫡 thanks for the review!
